services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/oss/SearchManagerImpl.java

```diff
@@ -162,6 +165,23 @@ public SearchResults search(UserInfo userInfo, SearchQuery searchQuery) {
     }
 
 
+    @Override
+    public SuggestionResults getSuggestion(UserInfo userInfo, SuggestionQuery suggestionQuery) {
+        SearchRequest searchRequest = OssUtil.generateSearchRequestForSuggestion(suggestionQuery);
+        try {
+            Map<String, List<Suggest<DocumentFields>>> suggestions =
+                    openSearchClient.search(searchRequest, DocumentFields.class).suggest();
+            SuggestionResults results = OssUtil.convertToSynapseSuggestionResult(suggestions);
+            SearchRequest aggregationRequest = OssUtil.generateAggregationRequestToLimitAccess(userInfo, results);
+            Map<String, Aggregate> aggregations = openSearchClient.search(aggregationRequest, DocumentFields.class).aggregations();
+            return OssUtil.eliminateSuggestionWithAccessDenied(results, aggregations);
```

Comment:
comment: filter the suggests to return only those which appear in document(s) the user can see